### PR TITLE
fixes issues about separator with visibility

### LINKED ISSUE #2606: [iOS] Ensure ToggleVisibility interacts with separators correctly
**Automatically generated task. Any updates to the title and body will be overwritten. Do NOT modify the labels.**

iOS fix of #2600

### PR DESCRIPTION
**Pull request by nesalang**
fixes #2606 

how verified:
used ToggleVisibility.AllElements.json.
by clicking the new action button, verified that separator is correctly toggled.

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleVisibilityTarget.mm [hunk 1]
```diff
@@ -36,17 +36,31 @@ - (instancetype)initWithActionElement:(std::shared_ptr<AdaptiveCards::ToggleVisi
...
         NSUInteger tag = hashString.hash;
         UIView *view = [_rootView viewWithTag:tag];
+        
+        NSMutableString *hashStringForSeparator = [NSMutableString stringWithCString:target->GetElementId().c_str() encoding:NSUTF8StringEncoding];
+        [hashStringForSeparator appendString:@"-separator"];
+        NSUInteger separatorTag = hashStringForSeparator.hash;
+        UIView *separator = [_rootView viewWithTag:separatorTag];
 
-        if(target->GetIsVisible() == AdaptiveCards::IsVisibleToggle){
+        if (target->GetIsVisible() == AdaptiveCards::IsVisibleToggle) {
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleVisibilityTarget.mm]: >   if (target->GetIsVisible() == [](start = 5, length = 32)

if you implement with a `switch` or use a local variable, you can avoid calling `GetIsVisible()` twice :)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRSeparator.mm [hunk 2]
```diff
@@ -109,49 +109,46 @@ + (void) renderActionsSeparator:(UIView *)view
...
+                         superview:(UIView *)view
+                        hostConfig:(std::shared_ptr<HostConfig> const &)config
+                           spacing:(Spacing)spacing
 {
     ACRSeparator *separator = nil;
     Spacing requestedSpacing = Spacing::None;
-    if(elem)
+    
+    if (elem)
     {
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRSeparator.mm]: >   { [](start = 1, length = 4)

nit: shouldn't this be on the line above?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm [hunk 3]
```diff
@@ -198,16 +198,21 @@ + (UIView *)render:(UIView<ACRIContentHoldingView> *)view
     UIView *prevStretchableElem = nil, *curStretchableElem = nil;
 
     auto firstelem = elems.begin();
-    for(const auto &elem:elems)
+    auto prevElem = *firstelem;
+    
+    for (const auto &elem:elems)
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm]: >: [](start = 25, length = 1)

nit: put spaces around the `:` here

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleVisibilityTarget.mm (hunk 1 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleVisibilityTarget.mm]: thanks

---
In reply to: [280537612](https://github.com/Microsoft/AdaptiveCards/pull/2829#discussion_r280537612) [](ancestors = 280537612)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm (hunk 3 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRenderer.mm]: thanks

---
In reply to: [280538099](https://github.com/Microsoft/AdaptiveCards/pull/2829#discussion_r280538099) [](ancestors = 280538099)

**Pull request merged**